### PR TITLE
Feature/site search with parameters

### PR DESCRIPTION
**Pull request by Primisen**

### CODE REVIEW

**Comment by vguzh** [comment]
please resolve conflicts

File: logstash/postgres_to_elasticsearch.conf [hunk 1]
```diff
@@ -0,0 +1,90 @@
+input {
+  jdbc {
+    jdbc_driver_library => "/usr/share/logstash/postgresql-42.7.3.jar"
+    jdbc_driver_class => "org.postgresql.Driver"
+    jdbc_connection_string => "jdbc:postgresql://db:5432/architecture-map?currentSchema=architecture_map"
+    jdbc_user => "${POSTGRES_USER}"
+    jdbc_password => "${POSTGRES_PASSWORD}"
+    schedule => "0 6 * * * America/Chicago"
```
Comment (vguzh) [inline logstash/postgres_to_elasticsearch.conf:8]: why America/Chicago?

File: src/main/kotlin/by/architecture_map/belarus/service/SearchService.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package by.architecture_map.belarus.service
+
+import by.architecture_map.belarus.entity.Construction
+
+interface SearchService {
+
+    fun constructionSearch(
+        architecturalStyleId: String?,
```
Comment (vguzh) [inline src/main/kotlin/by/architecture_map/belarus/service/SearchService.kt:8]: if the amount of parameters is too big, it is better to use DTO
but up too 5 parameters is acceptable

File: src/main/kotlin/by/architecture_map/belarus/service/impl/SearchServiceImpl.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        if (!region.isNullOrEmpty()) {
+            criteria = criteria.and("address.region").`is`(region)
+        }
+
+        if (!district.isNullOrEmpty()) {
+            criteria = criteria.and("address.district").`is`(district)
+        }
+
+        if (!buildingTimeFrom.isNullOrEmpty() && !buildingTimeTo.isNullOrEmpty()) {
```
Comment (vguzh) [inline src/main/kotlin/by/architecture_map/belarus/service/impl/SearchServiceImpl.kt:37]: why can't we use from and to independantly?

File: logstash/postgres_to_elasticsearch.conf (hunk 1 above)
Comment (Primisen) [reply logstash/postgres_to_elasticsearch.conf:8]: It is not really important in which time we will run script, the main that it runs every day. So, I was go the simplest way and used time from documentation :)

https://www.elastic.co/guide/en/logstash/current/plugins-inputs-jdbc.html

File: src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt [hunk 4]
```diff
@@ -12,40 +12,53 @@ class ConstructionServiceImpl(
...
+                buildingCentury = updatedConstruction.buildingCentury
+                images = updatedConstruction.images
+            }
+            .let {
+                constructionRepository.save(it)
+            }
+
+    override fun patchUpdate(id: Int, updatedConstruction: Construction): Construction =
+        constructionRepository.findById(id)
+            .orElseThrow { NotFoundException("Connection not found with id: $id") }
```
Comment (vguzh) [inline src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt:40]: are you sure that the comment is correct here?

File: src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt [hunk 5]
```diff
@@ -12,40 +12,53 @@ class ConstructionServiceImpl(
...
-        }
+                buildingCentury = updatedConstruction.buildingCentury
+                images = updatedConstruction.images
+            }
+            .let {
+                constructionRepository.save(it)
+            }
+
+    override fun patchUpdate(id: Int, updatedConstruction: Construction): Construction =
+        constructionRepository.findById(id)
```
Comment (vguzh) [inline src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt:39]: why have you removed the previous `applyUpdates` method or at least do not reuse `fun find(id: Int)`?

**Review by vguzh — CHANGES_REQUESTED** [review]

File: src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt (hunk 5 above)
Comment (Primisen) [reply src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt]: I'm surprised too. Apparently, I am don't correctly resolved merge conflicts. I am sorry(

**Review by vguzh — APPROVED** [review]

**Pull request merged**